client/main/client.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from os import mkdir
+from os import path
+from random import sample
+from string import ascii_lowercase
+from string import ascii_uppercase
+from string import digits
+import subprocess
+import sys
+
+from requests import get
+from requests import post
+from requests import RequestException
+
+from ...systemcheck.systemcheck import systemcheck
+
+EXIT_OK = 0
+
+SERVER_ADDRESS = "http://private-a6e53-dpcs.apiary-mock.com/"
+
+
+def generate_report(exit_code, stderr_output):
+    """Generate a report which will be sent to the DCPS server.
+
+    Arguments
+    ---------
+    exit_code : int
+        The exit code of the failed process
+
+    stderr_output : string
+        The output written on stderr by the failed process
+
+    Return
+    ------
+    template : dict
+        The report that can be sent to the DCPS server.
+    """
+    template = {
+        "crash_report": {
+            "application": {
+                # TO DO - implementation depends on the way how the demon is
+                # implemented
+            },
+            "system_info": {},
+            "exit_code": exit_code,
+            "stderr_output": stderr_output
+        }
+    }
+
+    system_info = systemcheck()
```

Comment:
As we're not sure yet how we are going to implement system info caching, this feature is not in the scope of this PR.